erpnext/stock/doctype/material_request/material_request.js

```diff
@@ -104,7 +104,26 @@ frappe.ui.form.on('Material Request', {
 					add_create_pick_list_button();
 					frm.add_custom_button(__("Transfer Material"),
 						() => frm.events.make_stock_entry(frm), __('Create'));
+
+					frm.add_custom_button(__("Begin Transit"),
```

Comment:
begin transit should be automatic or a checkbox as per my earlier suggestion so that it is API friendly